app/services/sets/product_set.rb

```diff
@@ -146,11 +146,11 @@ def count_result(saved)
 
     def notify_bugsnag(error, product, variant, variant_attributes)
       Bugsnag.notify(error) do |report|
-        report.add_metadata(:product, product.attributes)
-        report.add_metadata(:product_error, product.errors.first) unless product.valid?
-        report.add_metadata(:variant_attributes, variant_attributes)
-        report.add_metadata(:variant, variant.attributes)
-        report.add_metadata(:variant_error, variant.errors.first) unless variant.valid?
+        report.add_metadata( :product_set,
+                             { product: product.attributes, variant_attributes:,
+                               variant: variant.attributes } )
+        report.add_metadata(:product_set, :product_error, product.errors.first) if !product.valid?
+        report.add_metadata(:product_set, :variant_error, variant.errors.first) if !variant.valid?
```

Comment:
I thought that rubocop would prefer `unless` over `if !`. 
I think it's clear either way here, so all good. 👍

Reply:
I would normally use `unless` but then it makes the line too long, and rubocop linting ended up triggering class has too many lines, so I compromised :smile: